### ISSUE TITLE
Record filenames for all checksum files.

### ISSUE THREAD

**Issue by Conchylicultor** [tensorflow/datasets#2617]
With https://github.com/tensorflow/datasets/pull/2601 and https://github.com/tensorflow/datasets/issues/2422, users can now manually download all datasets. However this requires the `checksums/my_dataset.txt` (legacy) and `my_datasets/checksums.tsv` (new) to record the filename.

This was only added recently, so most checksums files do not contain the checksums. We should add a new `scripts/cleanup/`, similar to https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/scripts/cleanup/url_status_checker.py

The script would query all datasets to record the name in the checksums. No need to download the full file to get the name. Just using this function on a `request.Request` object should be enough: https://github.com/tensorflow/datasets/blob/ad114d874b9b92d6474b44e0aa92bc918a9b3b94/tensorflow_datasets/core/download/downloader.py#L90

`tfds.download.checksums.load_url_infos` and `tfds.download.checksums.save_url_infos` can be used to load/save the files:

https://github.com/tensorflow/datasets/blob/ad114d874b9b92d6474b44e0aa92bc918a9b3b94/tensorflow_datasets/core/download/checksums.py#L149

Once the script has been merged (PR1), then all checksums files could be updated (PR2):

Before:

```txt
<url> <checksums> <size>
```
After:
```txt
<url> <checksums> <size> <filename>
```

**Comment by harsh020** [comment]
Hi, I would like to work on this issue. I will open PR1 tonight (21 Oct).

Thanks.

**Comment by harsh020** [comment]
Hi, will open PR2  as soon as PR1 gets merged.

Thanks.

**Comment by Conchylicultor** [comment]
Yes, will merge this soon.
I just realise they might be an issue with drive.com url which might not be properly extracted. I'll fix those bugs internally, so I'll probably send PR2 myself.

**Comment by vijayphoenix** [comment]
FIxed by 24211da0
